docs/upgrade-guides/nextjs/v6.mdx

```diff
@@ -0,0 +1,120 @@
+---
+title: Upgrade to @clerk/nextjs v6
+description: Learn how to upgrade your application to use the latest Clerk Next.js SDK with support for Next.js 15.
+---
+
+To upgrade to `@clerk/nextjs` v6, use the upgrade CLI:
+
+```bash {{ filename: 'terminal' }}
+npx @clerk/upgrade
+```
+
+The CLI will walk you through the upgrade and apply changes to your application automatically.
+
+## `auth()` is now async
+
+To support Next.js's change to async dynamic APIs, `auth()` is now asynchronous. The codemod will migrate this for you.
+
+```diff
+import { auth } from '@clerk/nextjs/server';
+
+- export function GET() {
++ export async function GET() {
+-  const { userId } = auth();
++  const { userId } = await auth();
+  return new Response(JSON.stringify({ userId }));
+}
+```
+
+### In `clerkMiddleware`
+
+Now that `auth()` is asynchronous, custom middleware handlers that rely on `auth()` must also now be converted to be asynchronous. The codemod will migrate this for you.
+
+```diff
+import { clerkClient, clerkMiddleware } from '@clerk/nextjs/server';
+import { NextResponse } from 'next/server';
+
+- export default clerkMiddleware((auth, request) => {
++ export default clerkMiddleware(async (auth, request) => {
+-  const resolvedAuth = auth();
++  const resolvedAuth = auth();
```

Comment:
This diff is doesn't have any changes?

![Screenshot 2024-10-22 at 12 45 03 PM](https://github.com/user-attachments/assets/0349f554-cf08-46be-a116-7178138a6ad6)